### PR TITLE
Automatically generate profile icons

### PR DESCRIPTION
**Pull request by Mikunj**
This PR automatically generates profile icons based on a users public key.
We also have added a way for users to set their display name when registering, however this is not yet fully styled.

**NOTE:** After this merge you will have to run `yarn install` again.

screen shot 2018-12-03 at 4 09 37 pm

### CODE REVIEW

File: app/profile_images.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  }
+}
+
+module.exports = {
+  getOrCreateImagePath,
+  getImagePath,
+  hasImage,
+  removeImage,
+  hashCode,
+};
```
Comment (sachaaaaa) [inline app/profile_images.js]: woops, new line!

File: js/views/standalone_registration_view.js [hunk 2]
```diff
@@ -47,7 +47,17 @@
       this.accountManager
         .registerSingleDevice(
           this.$('#mnemonic').val(),
-          this.$('#mnemonic-language').val()
+          this.$('#mnemonic-language').val(),
+          async (pubKey) => {
+            await storage.setProfileName(this.$('#display-name').val());
+
+            // Update the conversation if we have it
+            const conversation = ConversationController.get(pubKey);
```
Comment (sachaaaaa) [inline js/views/standalone_registration_view.js]: Is there a case where there is an existing conversation with ourself upon account creation?

File: package.json [hunk 3]
```diff
@@ -118,7 +119,7 @@
     "bower": "^1.8.2",
     "chai": "^4.1.2",
     "electron": "2.0.8",
-    "electron-builder": "^20.13.5",
+    "electron-builder": "20.13.5",
```
Comment (sachaaaaa) [inline package.json]: was it breaking with any newer min-version?

Comment (sachaaaaa) [reply package.json]: Good to know!

**Comment by sachaaaaa** [comment]
Better with that last commit. I'll give it a few hours for @BeaudanBrown to have a look before merging, but to me it looks fine!

**Comment by Mikunj** [comment]
Hold off on merging, i'm looking into switching the libraries so that the profile pictures are more closer to github images

**Pull request merged**